dvc/stage/utils.py

```diff
@@ -330,7 +330,7 @@ def validate_state(
 def validate_kwargs(single_stage: bool = False, fname: str = None, **kwargs):
     """Prepare, validate and process kwargs passed from cli"""
     cmd = kwargs.get("cmd")
-    if not cmd:
+    if not cmd and not single_stage:
```

Comment:
not sure if this should be a separate `data_source` kwarg instead of using `single_stage`? @skshetry

Reply:
We still support stages in `.dvc` file, so it's not just about `data_source`. Maybe in 3.0? 😄